syntax = "proto2";

service Bank {
    rpc balance (balance_request) returns (balance_respose);
    rpc payment (payment_request) returns (payment_respose);
    rpc transfer (transfer_request) returns (transfer_respose);
    rpc end_of_work (end_of_work_request) returns (end_of_work_respose);

}

message balance_request {
    required string wallet_key = 1;
}

message balance_respose {
    required int32 balance_value = 1; 
}

message payment_request {
    required string wallet_key = 1;
    required int32 value = 2;
}

message payment_respose {
    required int32 status = 1; 
    optional fixed32 auth_token = 2; 
}

message transfer_request {
    required int32 calue = 1; 
    optional fixed32 auth_token = 2; 
    required string wallet_key = 3;
}

message transfer_respose {
    required int32 status = 1; 
}

message end_of_work_request {
}

message end_of_work_respose {
    required int32 status = 1; 
}